refactor: remove asset value from callback outputs (#3513)
0xMiden/protocol: docs/src/asset.md

#### Callback interfaces

> The transaction kernel invokes the callback on the issuing faucet as a validation hook. The callback receives the asset ID and value, but cannot replace the value: it either completes successfully or aborts the transaction. The kernel retains the original asset value and uses it after the callback.

Reviewer: ```suggestion
The transaction kernel invokes the callback on the issuing faucet as a validation hook. The callback receives the asset ID and value for inspection. The callback either completes successfully or aborts the transaction.
```

nit: I think this explains the change from this PR.
